core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadWriteLDAPUserStoreManager.java

```diff
@@ -917,6 +918,80 @@ protected void doSetUserAttributes(String userName, Map<String, String> processe
         }
     }
 
+    protected void doSetUserAttributes(String userName, Map<String, List<String>> claimAttributesToAdd,
+                                       Map<String, List<String>> claimAttributesToDelete,
+                                       Map<String, List<String>> claimAttributesToReplace, String profileName)
+            throws UserStoreException, NotImplementedException {
+
+        // Get the LDAP Directory context.
+        DirContext dirContext = this.connectionSource.getContext();
+        DirContext subDirContext = null;
+        // Search the relevant user entry by user name.
+        String userSearchBase = realmConfig.getUserStoreProperty(LDAPConstants.USER_SEARCH_BASE);
+        String userSearchFilter = realmConfig
+                .getUserStoreProperty(LDAPConstants.USER_NAME_SEARCH_FILTER);
+        // If user name contains domain name, remove domain name.
+        userName = UserCoreUtil.removeDomainFromName(userName);
+        userSearchFilter = userSearchFilter.replace("?", escapeSpecialCharactersForFilter(userName));
+
+        SearchControls searchControls = new SearchControls();
+        searchControls.setSearchScope(SearchControls.SUBTREE_SCOPE);
+        searchControls.setReturningAttributes(null);
+
+        NamingEnumeration<SearchResult> returnedResultList = null;
+        String returnedUserEntry = StringUtils.EMPTY;
+
+        try {
+            subDirContext = (DirContext) dirContext.lookup(escapeDNForSearch(userSearchBase));
+            returnedResultList = dirContext.search(escapeDNForSearch(userSearchBase), userSearchFilter, searchControls);
+
+            // Assume only one user is returned from the search.
+            if (returnedResultList.hasMore()) {
+                returnedUserEntry = returnedResultList.next().getName();
+                handleLdapUserNameAttributeChanges(claimAttributesToReplace, subDirContext, returnedUserEntry);
+            }
+
+        } catch (NamingException e) {
+            String errorMessage = "Results could not be retrieved from the directory context for user : " + userName;
+
```

Comment:
Shall we remove extra newlines?

Reply:
addressed in https://github.com/wso2/carbon-kernel/pull/2705/commits/fb430db0be811baeb1b5fbbd5e15c0cc8abd2c54